cooperator/tests/test_cooperator.py

```diff
@@ -64,104 +75,85 @@ def _pay_invoice(self, invoice, payment_date=None):
         )
         register_payments.create_payments()
 
+    @users("user-cooperator")
     def test_register_payment_for_capital_release(self):
-        self.as_cooperator_user()
-        self.request.validate_subscription_request()
-        invoice = self.request.capital_release_request
+        self.subscription_request_1.validate_subscription_request()
+        invoice = self.subscription_request_1.capital_release_request
 
         self._pay_invoice(invoice)
-        self.assertEqual(self.request.capital_release_request.state, "paid")
+        self.assertEqual(invoice.state, "paid")
 
-        partner = self.request.partner_id
+        partner = self.subscription_request_1.partner_id
         self.assertFalse(partner.coop_candidate)
         self.assertTrue(partner.member)
         self.assertTrue(partner.share_ids)
         self.assertEqual(partner.effective_date, Date.today())
 
         share = partner.share_ids[0]
-        self.assertEqual(share.share_number, self.request.ordered_parts)
-        self.assertEqual(share.share_product_id, self.request.share_product_id)
+        self.assertEqual(share.share_number, self.subscription_request_1.ordered_parts)
+        self.assertEqual(
+            share.share_product_id, self.subscription_request_1.share_product_id
+        )
         self.assertEqual(share.effective_date, Date.today())
 
-    def test_effective_date_from_payment_date(self):
-        self.as_cooperator_user()
-        self.request.validate_subscription_request()
-        invoice = self.request.capital_release_request
-        self._pay_invoice(invoice, datetime.date(2022, 6, 21))
-
-        partner = self.request.partner_id
-        self.assertEqual(partner.effective_date, datetime.date(2022, 6, 21))
```

Comment:
please don’t remove this test.

Reply:
I think i got confused in a messy rebase. Sorry and thanks for the catch.